tests/e2e/svmotion_detached_volume.go

```diff
@@ -435,10 +448,883 @@ var _ = ginkgo.Describe("[csi-block-vanilla] [csi-block-vanilla-parallelized] Re
+		ginkgo.By("verify we can read and write on the PVCs")
+		pods := createMultiplePods(ctx, client, pvclaims2d, true)
+		defer func() {
+			ginkgo.By("Delete the pod created")
+			deletePodsAndWaitForVolsToDetach(ctx, client, pods, true)
+		}()
+
+		ginkgo.By("Verify if VolumeID is created on the given datastores")
+		dsUrlWhereVolumeIsPresent := fetchDsUrl4CnsVol(e2eVSphere, volumeID)
+		framework.Logf("Volume: %s is present on %s", volumeID, dsUrlWhereVolumeIsPresent)
+		e2eVSphere.verifyDatastoreMatch(volumeID, datastoreUrls)
+
+		// Get the destination ds url where the volume will get relocated
+		destDsUrl := ""
+		for _, dsurl := range datastoreUrls {
+			if dsurl != dsUrlWhereVolumeIsPresent {
+				destDsUrl = dsurl
+				break
+			}
+		}
+
+		framework.Logf("dest url: %s", destDsUrl)
+		dsRefDest := getDsMoRefFromURL(ctx, destDsUrl)
+
+		rand.New(rand.NewSource(time.Now().Unix()))
+		testdataFile := fmt.Sprintf("/tmp/testdata_%v_%v", time.Now().Unix(), rand.Intn(1000))
+		ginkgo.By(fmt.Sprintf("Creating a 100mb test data file %v", testdataFile))
+		op, err := exec.Command("dd", "if=/dev/urandom", fmt.Sprintf("of=%v", testdataFile),
+			"bs=1M", "count=100").Output()
+		fmt.Println(op)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			op, err = exec.Command("rm", "-f", testdataFile).Output()
+			fmt.Println(op)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		originalSizeInMb, err := getFSSizeMb(f, pods[0])
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		snaps := []*snapV1.VolumeSnapshot{}
+		snapIDs := []string{}
+
+		//Get snapshot client using the rest config
+		restConfig := getRestConfigClient()
+		snapc, err := snapclient.NewForConfig(restConfig)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Create volume snapshot class")
+		volumeSnapshotClass, err := snapc.SnapshotV1().VolumeSnapshotClasses().Create(ctx,
+			getVolumeSnapshotClassSpec(snapV1.DeletionPolicy("Delete"), nil), metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("Volume snapshot class with name %q created", volumeSnapshotClass.Name)
+
+		defer func() {
+			err := snapc.SnapshotV1().VolumeSnapshotClasses().Delete(
+				ctx, volumeSnapshotClass.Name, metav1.DeleteOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Start relocation of volume to a different datastore")
+		err = e2eVSphere.cnsRelocateVolume(e2eVSphere, ctx, volumeID, dsRefDest)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Create a snapshot of volume and writing IO to pod while relocating volume")
+		ch := make(chan *snapV1.VolumeSnapshot)
+		lock := &sync.Mutex{}
+		var wg sync.WaitGroup
+		wg.Add(2)
+		go writeKnownData2PodInParallel(f, pods[0], testdataFile, &wg, originalSizeInMb-spareSpace)
+		go createSnapshotInParallel(ctx, namespace, snapc, pvclaim.Name, volumeSnapshotClass.Name,
```

Comment:
wait for a few secs before taking a snap, to let some IO to flow in

Reply:
Done